File: Core/SmartGrid/Models/Incident.cs
using SmartGrid.Models.Enumerations;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace SmartGrid.Models
{
    public class Incident
    {
        //public IncidentBasicInfo BasicInfo { get; set; }
        //BasicInfo
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        [Required]
        public IncidentType Type { get; set; }
        public int Priority { get; set; }
        public bool Confirmed { get; set; }
        public string Status { get; set; }
        public string Description { get; set; }
        [Required]
        public DateTime ETA { get; set; }
        [Required]
        public DateTime ATA { get; set; }
        [Required]
        public DateTime OutageTime { get; set; }
        [Required]
        public DateTime ETR { get; set; }
        public int AffectedConsumersNumber { get; set; }
        public int Calls { get; set; }
        public double Voltage { get; set; }
        [Required]
        public DateTime ScheduledTime { get; set; }

        //Devices
        public ICollection<Device> Devices { get; set; }

        //Resolution
        [Required]
        public string Cause { get; set; }
        [Required]
        public string SubCause { get; set; }
        [Required]
        public string ConstructionType { get; set; }
        [Required]
        public string Material { get; set; }

        //Calls
        public ICollection<Call> CallsList { get; set; } = new List<Call>();

        //Crew
        public string Crew { get; set; }

        //Multimedia
        public int MultimediaAttachment { get; set; }
    }
}

File: clientside/src/app/entities/incident.ts
import { Device } from "./device";

export class Incident {
    id: number;
    type: string;
    priority: number;
    confirmed: boolean;
    status: string;
    description: string;
    eta: Date;
    etaStr: string;
    ata: Date;
    ataStr: string;
    outageTime: Date;
    outageTimeStr: string;
    etr: Date;
    etrStr: string;
    affectedConsumersNumber: number;
    calls: number;
    voltage: number;
    scheduledTime: Date;
    scheduledTimeStr: string;
    devices: Array<Device>;
    cause: string;
    subCause: string;
    constructionType: string;
    material: string;
    //callsList: Array<Call>;
    crew: string;
    multimediaAttachment: number;

    constructor(type: string = "", priority: number = 3, confirmed: boolean = true, status: string = "", description: string = "",
        eta: Date = new Date(2021,6,15,13,15,0), ata: Date  = new Date(2021,6,15,13,15,0), outageTime: Date  = new Date(2021,6,15,13,15,0),
        etr: Date = new Date(2021,6,15,13,15,0), affectedConsumersNumber: number = 0,
        calls: number = 0, voltage: number = 0, scheduledTime: Date = new Date(2021,6,15,13,15,0), cause: string = "", subCause: string = "", 
        constructionType: string = "", material: string = "", crew: string = "", multimediaAttachment: number = 0){
            this.type = type;
            this.priority = priority;
            this.confirmed = confirmed;
            this.status = status;
            this.description = description;
            this.eta = eta;
            this.ata = ata;
            this.outageTime = outageTime;
            this.etr = etr;
            this.affectedConsumersNumber = affectedConsumersNumber;
            this.calls = calls;
            this.voltage = voltage;
            this.scheduledTime = scheduledTime;
            this.cause = cause;
            this.subCause = subCause;
            this.constructionType = constructionType;
            this.material = material;
            this.crew = crew;
            this.multimediaAttachment = multimediaAttachment;
        }
}


